lib/checkPullRequestLabels.js

```diff
@@ -48,33 +159,36 @@ module.exports.checkAssignee = async function(context) {
       // for invalid labels when a pull request is opened.
       return;
     }
-    var labelSubstrings = changelogLabel.split('@');
-    var projectOwner = labelSubstrings[labelSubstrings.length - 1].trim();
 
-    var assignees = [];
-    for (var i = 0; i < pullRequest.assignees.length; i++) {
-      assignees.push(pullRequest.assignees[i].login);
-    }
+    const projectOwner = getProjectOwner(changelogLabel);
 
-    if (
-        projectOwner !== pullRequest.user.login &&
-        !assignees.includes(projectOwner)) {
-      var assigneeParams = context.issue({ assignees: [projectOwner] });
-      await context.github.issues.addAssignees(assigneeParams);
+    const shouldAssignProjectOwner = canAssignProjectOwner(
+      pullRequest, changelogLabel);
+    if (shouldAssignProjectOwner) {
+      const assignees = pullRequest.assignees.map(assignee => assignee.login);
+      if(!assignees.includes(projectOwner)) {
+        var assigneeParams = context.issue({ assignees: [projectOwner] });
+        await context.github.issues.addAssignees(assigneeParams);
 
-      var commentParams = context.issue({
-        body:
-          'Assigning @' +
-          projectOwner +
-          ' for the first-pass review of this pull request. Thanks!',
-      });
-      await context.github.issues.createComment(commentParams);
+        var commentParams = context.issue({
+          body:
+            'Assigning @' +
+            projectOwner +
+            ' for the first-pass review of this pull request. Thanks!',
+        });
+        await context.github.issues.createComment(commentParams);
+      }
+      // Project owner is already assigned so we do nothing.
+    } else {
+      // Attempt to assign all codeowners.
+      await assignAllCodeowners(context);
     }
   }
   // If no changelog label is found, no action is required.
 };
 
 /**
+ * Check that changelog label is added to the PR.
```

Comment:
Update docstring as a noun. Here and below.

Reply:
Thanks for adding all these!

Reply:
Done.